tests/test_transform.py

```diff
@@ -180,10 +296,126 @@ def checkTransform(self, fromName, toName):
+                rawInPoint = makeRawPointData(nIn)
+                inPoint = fromEndpoint.pointFromData(rawInPoint)
+                outPoint = transform.tranForward(inPoint)
+                rawOutPoint = toEndpoint.dataFromPoint(outPoint)
+                # TODO replace all use of np.allclose with assertFloatsAlmostEqual once DM-9707 is fixed
+                assert_allclose(rawOutPoint, polyMap.tranForward(rawInPoint), err_msg=msg)
+                assert_allclose(rawOutPoint, frameSet.tranForward(rawInPoint), err_msg=msg)
+
+                # inverse transformation of one point;
+                # remember that the inverse will not give the original values!
+                inversePoint = transform.tranInverse(outPoint)
+                rawInversePoint = fromEndpoint.dataFromPoint(inversePoint)
+                assert_allclose(rawInversePoint, polyMap.tranInverse(rawOutPoint), err_msg=msg)
+                assert_allclose(rawInversePoint, frameSet.tranInverse(rawOutPoint), err_msg=msg)
+
+                # forward transformation of an array of points
+                nPoints = 7  # arbitrary
+                rawInArray = makeRawArrayData(nPoints, nIn)
+                inArray = fromEndpoint.arrayFromData(rawInArray)
+                outArray = transform.tranForward(inArray)
+                rawOutArray = toEndpoint.dataFromArray(outArray)
+                self.assertFloatsAlmostEqual(rawOutArray, polyMap.tranForward(rawInArray), msg=msg)
+                self.assertFloatsAlmostEqual(rawOutArray, frameSet.tranForward(rawInArray), msg=msg)
+
+                # inverse transformation of an array of points;
+                # remember that the inverse will not give the original values!
+                inverseArray = transform.tranInverse(outArray)
+                rawInverseArray = fromEndpoint.dataFromArray(inverseArray)
+                self.assertFloatsAlmostEqual(rawInverseArray, polyMap.tranInverse(rawOutArray), msg=msg)
+                self.assertFloatsAlmostEqual(rawInverseArray, frameSet.tranInverse(rawOutArray), msg=msg)
+
+                # If frameSet has a SkyFrame as base or current, try permuting that (in place);
+                # the Transform constructor should undo the permutation.
+                # Warning: you must permute the axes of the SkYFrame once it is in the frameSet,
+                # rather than before adding it to the FrameSet, else the behavior is wrong
+                for permFrameSet, isBasePermuted, isCurrPermuted in permFrameSetIter(frameSet):
+                    if isBasePermuted:
+                        self.assertEqual(permFrameSet.getFrame(astshim.FrameSet.BASE).getLonAxis(), 2)
+                    if isCurrPermuted:
+                        self.assertEqual(permFrameSet.getFrame(astshim.FrameSet.CURRENT).getLonAxis(), 2)
+                    transformBasePerm = transformClass(permFrameSet)
+                    # the base frame set in the transform should NOT be permuted
```

Comment:
I don't understand this comment. It looks below like you're requiring that neither the base nor the current frame be permuted.